net/dns/manager.go

```diff
@@ -79,6 +84,23 @@ func NewManager(logf logger.Logf, oscfg OSConfigurator, health *health.Tracker,
 		knobs:    knobs,
 		goos:     goos,
 	}
+
+	// Rate limit our attempts to correct our DNS configuration.  The recovery func is triggered
+	// by SERVFAIL errors, which can be occurring in batches at relatively high frequency.
+	limiter := rate.NewLimiter(1.0/5.0, 1)
+
+	// This will recompile the DNS config, which in turn will requery the system
+	// DNS settings.
+	m.resolver.SetServFailRecovery(func() {
+		if m.config == nil {
```

Comment:
do we not need some kind of synchronization here? if m.config is changing from local goroutines, and being called by the resolver I'd expect we'd need a lock?

Reply:
Done.